node/impl/full/eth.go

```diff
@@ -922,62 +825,135 @@ func (a *EthModule) Web3ClientVersion(ctx context.Context) (string, error) {
 	return build.UserVersion(), nil
 }
 
-func (a *EthModule) ethCallToFilecoinMessage(ctx context.Context, tx ethtypes.EthCall) (*types.Message, error) {
-	var from address.Address
-	if tx.From == nil || *tx.From == (ethtypes.EthAddress{}) {
-		// Send from the filecoin "system" address.
-		var err error
-		from, err = (ethtypes.EthAddress{}).ToFilecoinAddress()
+func (a *EthModule) TraceBlock(ctx context.Context, blkNum string) (interface{}, error) {
+	ts, err := getTipsetByBlockNr(ctx, a.Chain, blkNum, false)
```

Comment:
nit: i personally dislike "abbreviated" names like this -- would prefer `getTipsetByBlockNumber`, but feel free to ignore.

Reply:
will change as other methods use Number instead of Nr